File: src/csharp/api/Services/IOrderService.cs
using Example.Api.Dtos;
using Example.Api.Dtos.Requests;
using Example.Api.Dtos.Responses;

namespace Example.Api.Services;

/// <summary>
/// Service interface for managing orders.
/// </summary>
public interface IOrderService
{
    /// <summary>
    /// Retrieves an order by its identifier.
    /// </summary>
    /// <param name="id">The unique identifier of the order.</param>
    /// <returns>The order DTO.</returns>
    Task<ApiResult<OrderDto>> GetOrderAsync(long id);

    /// <summary>
    /// Creates a new order.
    /// </summary>
    /// <param name="request">The creation request.</param>
    /// <returns>The created order DTO.</returns>
    Task<ApiResult<OrderDto>> CreateOrderAsync(CreateOrderRequest request);

    /// <summary>
    /// Updates the message of an existing order.
    /// </summary>
    /// <param name="id">The unique identifier of the order.</param>
    /// <param name="message">The new message.</param>
    /// <returns>The updated order DTO.</returns>
    Task<ApiResult<OrderDto>> UpdateMessageAsync(long id, string message);
}

File: src/nodejs/src/services/IOrderService.ts
import { ApiDataResult } from "../dtos/ApiResult";
import { CreateOrderRequest } from "../dtos/CreateOrderRequest";
import { OrderDto } from "../dtos/OrderDto";

/**
 * Interface for order service operations.
 */
export interface IOrderService {
  /**
   * Retrieves an order by its ID.
   * @param id - The order ID.
   * @returns An API result containing the order if found, null otherwise.
   */
  getOrderAsync(id: number): Promise<ApiDataResult<OrderDto | null>>;

  /**
   * Creates a new order.
   * @param request - The create order request.
   * @returns An API result containing the created order.
   */
  createOrderAsync(request: CreateOrderRequest): Promise<ApiDataResult<OrderDto>>;

  /**
   * Updates an order's message.
   * @param id - The order ID.
   * @param message - The new message.
   * @returns An API result containing the updated order.
   */
  updateMessageAsync(id: number, message: string): Promise<ApiDataResult<OrderDto>>;
}
